pypesto/optimize/optimizer.py

```diff
@@ -630,3 +636,162 @@ def minimize(
 
     def is_least_squares(self):
         return False
+
+
+class NLoptOptimizer(Optimizer):
+    """
+    Global/Local optimization using NLopt.
+    Package homepage: https://nlopt.readthedocs.io/en/latest/
+    """
+
+    def __init__(self, method=None, local_method=None, options: Dict = None,
+                 local_options: Dict = None):
+        """
+        Parameters
+        ----------
+        method:
+            Local or global Optimizer to use for minimization.
+
+        local_method:
+            Local method to use in combination with the global optimizer (
+            for the MLSL family of solvers) or to solve a subproblem (for the
+            AUGLAG family of solvers)
+
+        options:
+            Optimizer options. scipy option `maxiter` is automatically
+            transformed into `maxeval` and takes precedence.
+
+        local_options:
+            Optimizer options for the local method
+        """
+
+        super().__init__()
+
+        if options is None:
+            options = {}
+        elif 'maxiter' in options:
+            options['maxeval'] = options.pop('maxiter')
+        if local_options is None:
+            local_options = {}
+        self.options = options
+        self.local_options = local_options
+        if nlopt is None:
+            raise ImportError(
+                "This optimizer requires an installation of NLopt. You can "
+                "install NLopt via pip install nlopt.")
+
+        if method is None:
+            method = nlopt.LD_LBFGS
+
+        needs_local_method = [
+            nlopt.G_MLSL, nlopt.G_MLSL_LDS, nlopt.GD_MLSL, nlopt.GD_MLSL_LDS,
+            nlopt.AUGLAG, nlopt.AUGLAG_EQ
+        ]
+
+        if local_method is None and method in needs_local_method:
+            local_method = nlopt.LD_LBFGS
+
+        if local_method is not None and method not in needs_local_method:
+            raise ValueError(f'Method "{method}" does not allow a local '
```

Comment:
Shall we rather raise a warning, that the local method is unused and continue without it?

Reply:
I see no good reason why a user would want that and I generally like working with sanitized input.